include/aspect/particle/output/interface.h

```diff
@@ -86,10 +86,37 @@ namespace aspect
                                const double current_time) = 0;
 
           /**
+           * Returns the particle file name taking into account the file index as well as the mpi proccess.
+           */
+          virtual
+          const std::string
+          get_file_name ();
```

Comment:
Right, but then you need to write the declaration as
```
  virtual std::string get_filename() const;
```
Are any of the derived classes going to actually implement a different behavior than the base class?

Also, please document the default behavior of the function.